libcontainerd/remote/client.go

```diff
@@ -502,27 +503,43 @@ func (c *container) createIO(fifos *cio.FIFOSet, stdinCloseSync chan containerd.
 
 	if io.Stdin != nil {
 		var (
-			err       error
+			closeErr  error
 			stdinOnce sync.Once
 		)
 		pipe := io.Stdin
 		io.Stdin = ioutils.NewWriteCloserWrapper(pipe, func() error {
 			stdinOnce.Do(func() {
-				err = pipe.Close()
-				// Do the rest in a new routine to avoid a deadlock if the
-				// Exec/Start call failed.
-				go func() {
-					p, ok := <-stdinCloseSync
+				closeErr = pipe.Close()
+
+				select {
+				case p, ok := <-stdinCloseSync:
 					if !ok {
 						return
 					}
-					err = p.CloseIO(context.Background(), containerd.WithStdinCloser)
-					if err != nil && strings.Contains(err.Error(), "transport is closing") {
-						err = nil
+					if err := closeStdin(context.Background(), p); err != nil {
+						if closeErr != nil {
+							closeErr = multierror.Append(closeErr, err)
```

Comment:
I wonder if we can replace this with stdlib functionality at some point, with the changes in go1.20; https://tip.golang.org/doc/go1.20#errors

> Wrapping multiple errors
>
> Go 1.20 expands support for error wrapping to permit an error to wrap multiple other errors.
>
> An error e can wrap more than one error by providing an Unwrap method that returns a []error.
>
> The errors.Is and errors.As functions have been updated to inspect multiply wrapped errors.
>
> The fmt.Errorf function now supports multiple occurrences of the %w format verb, which will cause it to return an error that wraps all of those error operands.
>
> The new function errors.Join returns an error wrapping a list of errors.